code/datums/saymode.dm

```diff
@@ -64,3 +65,26 @@
 		AI.holopad_talk(message, language)
 		return FALSE
 	return TRUE
+
+/datum/saymode/slime_link
+	key = MODE_KEY_SLIMELINK
+	mode = MODE_SLIMELINK
+	early = TRUE
+
+/datum/saymode/slime_link/handle_message(mob/living/user, message, datum/language/_language)
+	. = FALSE
+	if(!user || !user.mind)
+		return TRUE
+	if(!length(message))
+		return TRUE
+	if(ishuman(user))
+		var/mob/living/carbon/human/h_user = user
+		if(isstargazer(h_user))
+			var/datum/species/oozeling/stargazer/stargazer = h_user.dna.species
+			stargazer.slime_chat(h_user, message)
+			return
+	var/datum/weakref/owner_ref = GLOB.slime_links_by_mind[user.mind]
```

Comment:
This part is when you get the slime link as non-stargazer, right? Would you put a comment?